github/VisualStudio
# Fix broken build after clean or fresh clone

### What this PR does

This PR fixes the build errors after `git clean -xdf` or fresh clone. The main fix was moving from `Microsoft.VSSDK.BuildTools.15.0.26201` to `15.8.3252`.

I've also removed the old `Microsoft.VisualStudio.Sdk.BuildTasks.14.0` package and hack (see #1726) to import it when launching Visual Studio 2015. This fixes some warnings but wasn't the main problem.

- Update from `Microsoft.VSSDK.BuildTools.15.0.26201` to `15.8.3252`
- Disable Threading and SDK Analyzers (need to fix errors in another PR)
- Remove hack to import `BuildTasks.14.0` when `StartProgram` targeting Visual Studio 2015
- Remove legacy `Sdk.BuildTasks.14.0` packages
- Remove left over `packages.config files from `<PackageReference>` projects

### How to test

1. Clone `https://github.com/github/VisualStudio`
2. Open in Visual Studio 2017
3. Build solution

![image](https://user-images.githubusercontent.com/11719160/45750925-9330a880-bc08-11e8-8074-13f573c5a2b7.png)

To confirm broken build before the update from `Microsoft.VSSDK.BuildTools.15.0.26201` to `15.8.3252`.

1. `git reset 9cbf07820 --hard`
2. `git clean -xdf`
3. Open in Visual Studio 2017
4. Build solution

Expect first build to fail, but later builds to pass.

### Related

- dotnet/sdk#2535: The target "VerifyTargetVersion" does not exist in the project
